sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -635,3 +636,18 @@ object DataSourceStrategy {
     }
   }
 }
+
+case class PushDownCol(name: String, dataType: DataType)
```

Comment:
Ur, do we need to have a new case class for this?

@HyukjinKwon . Is this aligned with your recommendation? This seems to add a new complexity.

Reply:
We need it for keeping the `dataType`.

Reply:
I see. what about doing it as:

```scala
    case expressions.Contains(e: Expression, Literal(v: UTF8String, StringType))
        if PushDownCol.unapply(name).isDefined =>
      e.dataType ...
      name ...
```

I faced similar problem before and worked around as above.

Reply:
at https://github.com/apache/spark/blob/0032d85153e34b9ac69598b7dff530094ed0f640/sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala#L245-L248

Reply:
WDYT now?